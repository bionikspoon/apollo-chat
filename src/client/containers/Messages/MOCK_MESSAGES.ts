// generated from https://www.mockaroo.com/
export default [
  {
    body:
      'suspendisse accumsan tortor quis turpis sed ante vivamus tortor duis mattis egestas metus aenean fermentum donec ut mauris eget massa tempor convallis nulla neque libero convallis eget eleifend luctus ultricies eu nibh quisque id justo sit amet sapien dignissim vestibulum vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae nulla dapibus dolor vel est donec odio justo sollicitudin ut suscipit a feugiat et eros vestibulum ac est lacinia nisi venenatis tristique fusce congue',
    id: 1,
    user: 'felis ut at dolor quis odio',
  },
  {
    body:
      'blandit ultrices enim lorem ipsum dolor sit amet consectetuer adipiscing elit proin interdum mauris non ligula pellentesque ultrices phasellus id sapien in sapien iaculis congue vivamus metus arcu adipiscing molestie',
    id: 2,
    user: 'vel augue vestibulum ante ipsum primis in faucibus orci',
  },
  {
    body:
      'nunc proin at turpis a pede posuere nonummy integer non velit donec diam neque vestibulum eget vulputate ut ultrices vel augue vestibulum ante ipsum primis in',
    id: 3,
    user: 'id nisl venenatis lacinia aenean',
  },
  {
    body:
      'morbi sem mauris laoreet ut rhoncus aliquet pulvinar sed nisl nunc rhoncus dui vel sem sed sagittis nam congue risus semper porta volutpat quam pede lobortis ligula sit amet eleifend pede libero quis orci nullam molestie nibh in lectus pellentesque at nulla suspendisse potenti cras in purus eu magna vulputate luctus cum sociis natoque penatibus et magnis dis parturient montes nascetur ridiculus mus vivamus vestibulum sagittis sapien cum sociis natoque penatibus et magnis dis parturient montes nascetur ridiculus mus etiam',
    id: 4,
    user: 'semper est quam',
  },
  {
    body:
      'purus eu magna vulputate luctus cum sociis natoque penatibus et magnis dis parturient montes nascetur ridiculus mus vivamus vestibulum sagittis sapien cum sociis natoque penatibus et magnis dis parturient montes nascetur ridiculus mus etiam vel augue vestibulum rutrum rutrum neque aenean auctor gravida sem praesent id massa id nisl venenatis lacinia aenean sit amet justo morbi ut odio cras mi pede malesuada in imperdiet et',
    id: 5,
    user: 'aliquam sit amet',
  },
  {
    body:
      'nam ultrices libero non mattis pulvinar nulla pede ullamcorper augue a suscipit nulla elit ac nulla sed vel enim sit amet nunc viverra dapibus nulla suscipit ligula in lacus curabitur at ipsum ac tellus semper interdum mauris ullamcorper purus sit amet nulla',
    id: 6,
    user: 'vestibulum quam sapien varius ut blandit non interdum in',
  },
  {
    body:
      'lectus pellentesque eget nunc donec quis orci eget orci vehicula condimentum curabitur in libero ut massa volutpat convallis morbi odio odio elementum eu interdum eu tincidunt in leo maecenas pulvinar lobortis est phasellus sit amet erat',
    id: 7,
    user: 'ut at dolor quis odio consequat varius integer',
  },
  {
    body:
      'dictumst morbi vestibulum velit id pretium iaculis diam erat fermentum justo nec condimentum neque sapien placerat ante nulla justo aliquam quis turpis eget elit sodales scelerisque mauris sit amet eros suspendisse accumsan tortor quis turpis sed ante vivamus tortor duis mattis egestas metus aenean',
    id: 8,
    user: 'odio consequat varius integer ac leo pellentesque ultrices mattis',
  },
  {
    body:
      'vestibulum sagittis sapien cum sociis natoque penatibus et magnis dis parturient montes nascetur ridiculus mus etiam vel augue vestibulum rutrum rutrum neque aenean auctor gravida sem praesent id massa id nisl venenatis lacinia aenean sit amet justo morbi ut odio cras mi pede malesuada in imperdiet et commodo vulputate justo in blandit ultrices enim lorem',
    id: 9,
    user: 'congue diam id ornare imperdiet sapien urna',
  },
  {
    body:
      'lacus at velit vivamus vel nulla eget eros elementum pellentesque quisque porta volutpat erat quisque',
    id: 10,
    user: 'ut odio cras mi pede malesuada in imperdiet',
  },
  {
    body:
      'curae duis faucibus accumsan odio curabitur convallis duis consequat dui nec nisi volutpat eleifend donec ut dolor morbi vel lectus in quam fringilla rhoncus mauris enim leo rhoncus sed vestibulum sit amet cursus id turpis integer',
    id: 11,
    user:
      'posuere cubilia curae duis faucibus accumsan odio curabitur convallis',
  },
  {
    body:
      'in libero ut massa volutpat convallis morbi odio odio elementum eu interdum eu tincidunt in leo maecenas',
    id: 12,
    user: 'risus praesent lectus vestibulum quam sapien varius',
  },
  {
    body:
      'ut suscipit a feugiat et eros vestibulum ac est lacinia nisi venenatis tristique fusce congue diam id ornare imperdiet sapien urna pretium nisl ut volutpat sapien arcu sed augue aliquam erat volutpat in congue etiam justo',
    id: 13,
    user: 'quam sollicitudin vitae consectetuer eget rutrum at',
  },
  {
    body:
      'sollicitudin ut suscipit a feugiat et eros vestibulum ac est lacinia nisi venenatis tristique fusce congue diam id ornare imperdiet sapien urna pretium nisl ut volutpat sapien arcu sed augue aliquam erat volutpat in congue etiam justo etiam',
    id: 14,
    user: 'amet lobortis sapien sapien non mi integer ac',
  },
  {
    body:
      'ultrices posuere cubilia curae mauris viverra diam vitae quam suspendisse potenti nullam porttitor lacus at turpis donec posuere metus vitae ipsum aliquam non mauris morbi non lectus aliquam sit amet diam in magna bibendum',
    id: 15,
    user: 'justo etiam pretium iaculis justo',
  },
  {
    body:
      'neque duis bibendum morbi non quam nec dui luctus rutrum nulla tellus in sagittis dui vel nisl duis ac nibh fusce lacus purus',
    id: 16,
    user: 'pede lobortis',
  },
  {
    body:
      'diam erat fermentum justo nec condimentum neque sapien placerat ante nulla justo aliquam quis turpis eget elit sodales scelerisque mauris sit amet eros suspendisse accumsan tortor quis turpis sed ante vivamus tortor duis mattis egestas metus aenean fermentum donec ut mauris eget massa tempor convallis nulla neque libero convallis eget eleifend luctus ultricies eu nibh quisque id justo sit amet sapien dignissim vestibulum vestibulum ante ipsum primis in faucibus orci luctus',
    id: 17,
    user: 'hendrerit at vulputate vitae nisl aenean lectus',
  },
  {
    body:
      'tincidunt nulla mollis molestie lorem quisque ut erat curabitur gravida',
    id: 18,
    user: 'ac neque duis',
  },
  {
    body:
      'dolor sit amet consectetuer adipiscing elit proin risus praesent lectus vestibulum quam sapien varius ut blandit non interdum in ante vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae duis faucibus accumsan odio curabitur convallis duis consequat dui nec nisi volutpat eleifend donec ut dolor morbi vel lectus in quam fringilla rhoncus mauris enim leo rhoncus sed vestibulum sit amet cursus id turpis integer aliquet massa id lobortis convallis tortor risus',
    id: 19,
    user: 'eu nibh quisque id justo sit amet',
  },
  {
    body:
      'libero ut massa volutpat convallis morbi odio odio elementum eu interdum eu',
    id: 20,
    user: 'placerat praesent blandit nam',
  },
  {
    body:
      'augue a suscipit nulla elit ac nulla sed vel enim sit amet nunc viverra dapibus nulla suscipit ligula in lacus curabitur at ipsum ac tellus semper interdum mauris ullamcorper purus sit amet nulla quisque arcu libero rutrum ac lobortis vel dapibus at diam nam',
    id: 21,
    user: 'nulla nunc purus phasellus in felis donec',
  },
  {
    body:
      'vulputate luctus cum sociis natoque penatibus et magnis dis parturient montes nascetur ridiculus mus vivamus vestibulum sagittis sapien cum sociis natoque penatibus et magnis dis parturient montes nascetur ridiculus mus etiam',
    id: 22,
    user: 'elit proin interdum',
  },
  {
    body:
      'nunc rhoncus dui vel sem sed sagittis nam congue risus semper porta volutpat quam pede lobortis ligula sit amet eleifend pede libero quis orci nullam molestie nibh in lectus pellentesque',
    id: 23,
    user: 'fermentum justo nec condimentum neque',
  },
  {
    body:
      'vestibulum proin eu mi nulla ac enim in tempor turpis nec euismod scelerisque quam turpis adipiscing lorem vitae mattis nibh ligula nec sem duis aliquam convallis nunc',
    id: 24,
    user: 'faucibus orci luctus et ultrices posuere cubilia curae mauris',
  },
  {
    body:
      'augue vestibulum rutrum rutrum neque aenean auctor gravida sem praesent id massa id nisl venenatis lacinia aenean sit amet justo morbi ut odio cras mi pede malesuada in imperdiet et commodo vulputate justo in blandit ultrices enim lorem ipsum dolor sit amet consectetuer adipiscing elit',
    id: 25,
    user: 'nisi nam ultrices libero non mattis pulvinar nulla',
  },
  {
    body:
      'quis lectus suspendisse potenti in eleifend quam a odio in hac habitasse platea dictumst maecenas ut massa quis augue luctus tincidunt nulla mollis molestie lorem quisque ut erat curabitur gravida nisi at nibh in',
    id: 26,
    user: 'libero quis orci nullam molestie nibh in lectus pellentesque at',
  },
  {
    body:
      'nisi at nibh in hac habitasse platea dictumst aliquam augue quam sollicitudin vitae',
    id: 27,
    user: 'arcu sed augue',
  },
  {
    body:
      'vestibulum aliquet ultrices erat tortor sollicitudin mi sit amet lobortis sapien sapien non mi integer ac neque duis bibendum morbi non quam nec dui luctus rutrum nulla tellus in sagittis dui vel nisl duis ac nibh fusce',
    id: 28,
    user: 'nulla elit ac nulla',
  },
  {
    body:
      'eleifend luctus ultricies eu nibh quisque id justo sit amet sapien dignissim vestibulum vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae nulla dapibus dolor vel est donec odio justo sollicitudin ut suscipit a feugiat et eros vestibulum ac est lacinia nisi venenatis tristique fusce congue diam id ornare imperdiet sapien urna pretium',
    id: 29,
    user: 'vulputate nonummy maecenas tincidunt lacus at velit vivamus vel',
  },
  {
    body:
      'et eros vestibulum ac est lacinia nisi venenatis tristique fusce congue diam id ornare imperdiet sapien urna pretium nisl ut volutpat sapien arcu sed augue aliquam erat volutpat in congue etiam justo etiam pretium iaculis justo in hac habitasse platea dictumst etiam faucibus cursus urna ut tellus nulla ut erat id mauris vulputate elementum nullam varius nulla facilisi cras non velit nec nisi',
    id: 30,
    user:
      'faucibus orci luctus et ultrices posuere cubilia curae donec pharetra',
  },
]
