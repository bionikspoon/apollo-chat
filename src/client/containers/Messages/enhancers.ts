import { Intent } from '@blueprintjs/core'
import { UpdateQueryFn } from 'apollo-client/core/watchQueryOptions'
import gql from 'graphql-tag'
import { append, evolve, pipe, prop, uniqBy } from 'ramda'
import { ChildDataProps, compose, graphql, MutationFunc } from 'react-apollo'

const MESSAGES_QUERY = gql`
  query Messages {
    messages {
      id
      body
      color
      date
      user
      isRead @client
    }
  }
`
const MESSAGE_ADDED_SUBSCRIPTION = gql`
  subscription Messages {
    messageAdded {
      id
      body
      color
      date
      user
      isRead @client
    }
  }
`

const READ_MESSAGE = gql`
  mutation ReadMessage($id: ID!, $isRead: Boolean!) {
    readMessage(id: $id, isRead: $isRead) @client
  }
`

export default compose(
  graphql<{}, IData, {}, IQueryChildProps>(MESSAGES_QUERY, {
    props: props => ({
      ...props,

      subscribeToMessages: () => {
        if (!props.data) return

        props.data.subscribeToMore({
          document: MESSAGE_ADDED_SUBSCRIPTION,
          updateQuery,
        })
      },
    }),
  }),
  graphql(READ_MESSAGE, { name: 'readMessage' })
)

export type MessagesProps = ChildDataProps<IQueryChildProps, IData> & {
  readMessage: MutationFunc<{}, { id: string; isRead: boolean }>
}

interface IData {
  messages: [
    {
      id: string
      body: string
      user: string
      date: string
      color: Intent
      isRead: boolean
    }
  ]
}
interface IQueryChildProps {
  subscribeToMessages: () => void
}

const addItem = (item: any) =>
  pipe(
    append(item),
    uniqBy(prop('id'))
  )

const updateQuery: UpdateQueryFn<any> = (
  previousResult,
  { subscriptionData }
) =>
  evolve({
    messages: addItem(subscriptionData.data.messageAdded),
  })(previousResult)
