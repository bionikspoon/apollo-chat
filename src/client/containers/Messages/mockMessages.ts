// generated from https://www.mockaroo.com/
export default [
  {
    body:
      'maecenas rhoncus aliquam lacus morbi quis tortor id nulla ultrices aliquet maecenas',
    id: 1,
    user: 'Kathrine',
  },
  {
    body:
      'vivamus in felis eu sapien cursus vestibulum proin eu mi nulla ac enim in tempor turpis nec euismod scelerisque quam turpis adipiscing lorem vitae mattis nibh ligula nec sem duis aliquam convallis nunc proin',
    id: 2,
    user: 'Claudian',
  },
  {
    body:
      'feugiat non pretium quis lectus suspendisse potenti in eleifend quam a odio in hac habitasse platea dictumst maecenas ut massa',
    id: 3,
    user: 'Wendell',
  },
  {
    body:
      'elit sodales scelerisque mauris sit amet eros suspendisse accumsan tortor quis turpis sed ante vivamus tortor duis mattis egestas metus aenean fermentum donec ut mauris eget massa tempor convallis nulla neque libero convallis eget eleifend luctus ultricies eu nibh quisque id justo sit amet sapien dignissim vestibulum vestibulum ante',
    id: 4,
    user: 'Elianora',
  },
  {
    body:
      'id nisl venenatis lacinia aenean sit amet justo morbi ut odio cras mi pede malesuada in imperdiet et commodo vulputate justo in blandit ultrices enim lorem ipsum dolor sit amet consectetuer adipiscing elit',
    id: 5,
    user: 'Bertina',
  },
  {
    body:
      'curae donec pharetra magna vestibulum aliquet ultrices erat tortor sollicitudin mi sit amet lobortis sapien sapien non mi integer ac',
    id: 6,
    user: 'Kalle',
  },
  {
    body:
      'eu orci mauris lacinia sapien quis libero nullam sit amet turpis elementum ligula vehicula consequat morbi a ipsum integer a nibh in quis justo maecenas rhoncus aliquam lacus morbi quis tortor id nulla ultrices aliquet',
    id: 7,
    user: 'Heidi',
  },
  {
    body:
      'in imperdiet et commodo vulputate justo in blandit ultrices enim lorem ipsum dolor sit amet consectetuer adipiscing elit',
    id: 8,
    user: 'Ivett',
  },
  {
    body:
      'amet consectetuer adipiscing elit proin interdum mauris non ligula pellentesque ultrices phasellus id sapien in sapien iaculis congue vivamus metus arcu adipiscing molestie hendrerit at vulputate vitae nisl aenean lectus pellentesque eget nunc',
    id: 9,
    user: 'Corine',
  },
  {
    body:
      'magna at nunc commodo placerat praesent blandit nam nulla integer pede justo lacinia eget tincidunt eget tempus vel pede morbi porttitor lorem id ligula suspendisse ornare consequat lectus in',
    id: 10,
    user: 'Carce',
  },
  {
    body: 'elit sodales scelerisque mauris sit amet eros suspendisse',
    id: 11,
    user: 'Emmy',
  },
  {
    body:
      'in faucibus orci luctus et ultrices posuere cubilia curae duis faucibus accumsan odio curabitur convallis duis consequat dui nec nisi volutpat eleifend donec ut dolor morbi vel lectus in quam fringilla rhoncus mauris enim',
    id: 12,
    user: 'Phylis',
  },
  {
    body: 'est risus',
    id: 13,
    user: 'Bevan',
  },
  {
    body:
      'at turpis a pede posuere nonummy integer non velit donec diam neque vestibulum eget vulputate ut ultrices vel augue vestibulum',
    id: 14,
    user: 'Sloan',
  },
  {
    body:
      'consequat varius integer ac leo pellentesque ultrices mattis odio donec vitae nisi nam ultrices libero non mattis pulvinar nulla pede ullamcorper augue a suscipit nulla elit ac nulla sed',
    id: 15,
    user: 'Nobe',
  },
  {
    body:
      'pulvinar sed nisl nunc rhoncus dui vel sem sed sagittis nam congue risus semper porta volutpat quam pede lobortis ligula sit amet eleifend pede',
    id: 16,
    user: 'Palm',
  },
  {
    body:
      'est risus auctor sed tristique in tempus sit amet sem fusce consequat nulla nisl nunc nisl duis bibendum felis sed interdum venenatis turpis enim',
    id: 17,
    user: 'Waldon',
  },
  {
    body:
      'ac leo pellentesque ultrices mattis odio donec vitae nisi nam ultrices',
    id: 18,
    user: 'Laureen',
  },
  {
    body:
      'arcu adipiscing molestie hendrerit at vulputate vitae nisl aenean lectus pellentesque eget',
    id: 19,
    user: 'Tris',
  },
  {
    body:
      'id consequat in consequat ut nulla sed accumsan felis ut at dolor quis odio consequat varius integer ac leo pellentesque ultrices mattis odio donec vitae nisi nam ultrices libero non mattis pulvinar nulla pede ullamcorper augue a suscipit nulla',
    id: 20,
    user: 'Glenda',
  },
  {
    body:
      'in hac habitasse platea dictumst morbi vestibulum velit id pretium iaculis diam erat fermentum justo nec condimentum neque sapien placerat ante nulla justo aliquam quis turpis eget elit sodales scelerisque mauris sit amet eros suspendisse accumsan tortor quis turpis sed ante vivamus',
    id: 21,
    user: 'Dalis',
  },
  {
    body: 'ac leo pellentesque ultrices mattis odio donec vitae nisi nam',
    id: 22,
    user: 'Denys',
  },
  {
    body:
      'et magnis dis parturient montes nascetur ridiculus mus etiam vel augue vestibulum rutrum rutrum neque aenean auctor gravida sem praesent id massa',
    id: 23,
    user: 'Aurore',
  },
  {
    body:
      'congue vivamus metus arcu adipiscing molestie hendrerit at vulputate vitae nisl aenean lectus pellentesque eget nunc donec quis orci eget orci vehicula condimentum curabitur in libero ut massa volutpat convallis morbi odio odio elementum eu interdum eu tincidunt in leo maecenas pulvinar',
    id: 24,
    user: 'Lula',
  },
  {
    body:
      'penatibus et magnis dis parturient montes nascetur ridiculus mus etiam vel augue vestibulum rutrum rutrum neque aenean auctor gravida sem praesent id massa id nisl venenatis lacinia aenean sit amet justo morbi ut odio cras mi pede malesuada',
    id: 25,
    user: 'Lewie',
  },
  {
    body:
      'eleifend quam a odio in hac habitasse platea dictumst maecenas ut massa quis augue luctus tincidunt nulla mollis molestie lorem quisque ut erat curabitur gravida nisi at nibh in hac habitasse platea dictumst aliquam',
    id: 26,
    user: 'Chrystel',
  },
  {
    body:
      'sed justo pellentesque viverra pede ac diam cras pellentesque volutpat dui maecenas tristique est et tempus semper est quam pharetra magna ac consequat metus sapien ut nunc vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae mauris viverra diam vitae quam suspendisse potenti nullam',
    id: 27,
    user: 'Cristabel',
  },
  {
    body:
      'vel enim sit amet nunc viverra dapibus nulla suscipit ligula in lacus curabitur at ipsum ac tellus semper interdum mauris ullamcorper purus sit amet nulla quisque arcu libero rutrum ac lobortis vel dapibus at diam nam tristique',
    id: 28,
    user: 'Costa',
  },
  {
    body:
      'libero non mattis pulvinar nulla pede ullamcorper augue a suscipit nulla elit ac nulla sed vel enim sit amet nunc viverra dapibus nulla suscipit ligula in lacus curabitur at ipsum ac tellus semper interdum mauris',
    id: 29,
    user: 'Rutherford',
  },
  {
    body:
      'nisi nam ultrices libero non mattis pulvinar nulla pede ullamcorper augue',
    id: 30,
    user: 'Rosanne',
  },
]
