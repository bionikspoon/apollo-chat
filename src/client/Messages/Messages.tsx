import { Tag } from '@blueprintjs/core'
import { css, StyleSheet } from 'aphrodite'
import { append, evolve, pipe, prop, reverse, uniqBy } from 'ramda'
import * as React from 'react'

import { MESSAGE_ADDED_SUBSCRIPTION, MessageProps } from './enhancers'

const addItem = (item: any) =>
  pipe(
    append(item),
    uniqBy(prop('id'))
  )

export default class Messages extends React.Component<MessageProps> {
  public componentDidMount() {
    const { props } = this
    if (!props.data) return

    props.data.subscribeToMore({
      document: MESSAGE_ADDED_SUBSCRIPTION,
      updateQuery: (previousResult, { subscriptionData }) =>
        evolve({
          messages: addItem(subscriptionData.data.messageAdded),
        })(previousResult),
    })
  }

  public render() {
    const { props } = this
    if (!props.data) return null
    if (props.data.loading) return <p>Loading...</p>
    if (props.data.error) return <p>Error :(</p>
    if (!props.data.messages) return null

    const messages = reverse(props.data.messages)

    return (
      <div className={css(styles.container)}>
        {messages.map(message => (
          <p key={message.id} className={css(styles.message)}>
            <Tag minimal={true} intent="primary">
              {message.user}
            </Tag>
            <span className={css(styles.messageBody)}>{message.body}</span>
          </p>
        ))}
      </div>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column-reverse',
    overflowY: 'scroll',
  },
  message: { marginBottom: '.2rem' },
  messageBody: { marginLeft: '.5rem' },
})
