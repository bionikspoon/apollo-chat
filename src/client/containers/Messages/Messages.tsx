import { css, StyleSheet } from 'aphrodite/no-important'
import { reverse } from 'ramda'
import * as React from 'react'
import { MessagesProps } from './enhancers'
import Loading from './Loading'
import Message from './Message'

export default class Messages extends React.Component<MessagesProps> {
  public componentDidMount() {
    if (!this.props.data) return
    this.props.subscribeToMessages()
  }

  public render() {
    const { props } = this

    if (props.data.loading) return <Loading css={styles.container} />
    if (!props.data.messages) return <Loading css={styles.container} />

    const messages = reverse(props.data.messages)

    return (
      <div className={css(styles.container)}>
        {messages.map(message => (
          <Message
            body={message.body}
            color={message.color}
            id={message.id}
            key={message.id}
            loading={false}
            user={message.user}
            isRead={message.isRead}
            onReadMessage={this.handleReadMessage}
          />
        ))}
      </div>
    )
  }

  private handleReadMessage = (id: string, isRead: boolean) => {
    const { props } = this
    props.readMessage({ variables: { id, isRead } })
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column-reverse',
    overflowY: 'scroll',
  },
})
