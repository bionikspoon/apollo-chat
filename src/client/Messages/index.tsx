import enhancer from './enhancers'
import Messages from './Messages'

export default enhancer(Messages)
