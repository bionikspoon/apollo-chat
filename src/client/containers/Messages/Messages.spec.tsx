import { shallow } from 'enzyme'
import React from 'react'
import Messages from '.'

test('it matches snapshot', () => {
  const wrapper = shallow(<Messages />)
  expect(wrapper).toMatchSnapshot()
})
