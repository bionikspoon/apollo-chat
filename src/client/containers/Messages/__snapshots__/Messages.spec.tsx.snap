// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<Query
  displayName="Apollo(Apollo(Messages))"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "Messages",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "messages",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "body",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "color",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "date",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "user",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [
                        Object {
                          "arguments": Array [],
                          "kind": "Directive",
                          "name": Object {
                            "kind": "Name",
                            "value": "client",
                          },
                        },
                      ],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "isRead",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 120,
        "start": 0,
      },
    }
  }
  skip={false}
  warnUnhandledError={true}
>
  <Component />
</Query>
`;
